src/views/studio/EditVideoSheet/EditVideoForm/EditVideoForm.tsx

```diff
@@ -224,44 +240,46 @@ export const EditVideoForm: React.FC<EditVideoFormProps> = ({
     setValue,
   ])
 
-  const handleSubmit = createSubmitHandler(async (data: EditVideoFormFields) => {
-    // do initial validation
-    if (!isEdit && !data.assets.video?.blob) {
-      setFileSelectError('Video file cannot be empty')
-      return
-    }
-    if (!data.assets.thumbnail?.url) {
-      setFileSelectError('Thumbnail cannot be empty')
-      return
+  const handleSubmit = createSubmitHandler(
+    async (data: EditVideoFormFields) => {
+      // do initial validation
+      if (!isEdit && !data.assets.video?.blob) {
+        setFileSelectError('Video file cannot be empty')
+        return
+      }
+      if (!data.assets.thumbnail?.url) {
+        setFileSelectError('Thumbnail cannot be empty')
+        return
+      }
+
+      const callback = () => {
+        setForceReset(true)
+      }
+
+      debouncedDraftSave.current.flush()
+
+      await onSubmit(data, dirtyFields, callback)
+    },
+    (errors) => {
+      const error = Object.values(errors)[0] as FieldError
+      const ref = error.ref as HTMLElement
+      ref.scrollIntoView({
+        behavior: 'smooth',
+        block: 'start',
+      })
     }
+  )
 
-    const callback = () => {
-      setForceReset(true)
+  watch((data) => {
```

Comment:
This will add a new callback on every render. You can easily get to a point at which this will be called 50 times on each key press :P

We need to put this inside some `useEffect`. Calling `watch` also returns a reference to subscription that we can use for cleanup if needed